src/button/index.tsx

```diff
@@ -9,26 +9,30 @@ import Loader from 'loader'
 import Circle from 'icon/circleIcon'
 import Check from 'icon/checkIcon'
 
+export enum ButtonStatus {
+  PRIMARY = 'primary',
+  SECONDARY = 'secondary',
+  WARNING = 'warning',
+  UNSTYLED = 'unstyled',
+  LOADING = 'loading',
+  CHECKED = 'checked',
+}
+
 export interface ButtonProps {
   readonly type?: string,
   readonly href?: string | JSX.Element,
   readonly children?: string | number | React.ReactNode,
   readonly className?: Classcat.Class,
   readonly title?: string,
-  readonly primary?: boolean,
-  readonly secondary?: boolean,
-  readonly loading?: boolean,
-  readonly valid?: boolean,
-  readonly warning?: boolean,
+  readonly status?: ButtonStatus,
```

Comment:
is it really optional?

Reply:
I put it optional to have a default value (which is primary, by a bit of an arbitrary decision). I agree that most of the times you'll precise it, but if you just want a simple Button without any big status changing, it's easier to write with a default value.